python/lsst/analysis/tools/analysisParts/rhoStatistics.py

```diff
@@ -0,0 +1,77 @@
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+from __future__ import annotations
+
+__all__ = ("RhoStatisticsMixin",)
+
+from ..actions.vector import (
+    CalcRhoStatistics,
+    CoaddPlotFlagSelector,
+    DownselectVector,
+    LoadVector,
+    MagColumnNanoJansky,
+    PsfStarSelector,
+    SnSelector,
+    VectorSelector,
+)
+from ..interfaces import AnalysisTool
+
+
+class RhoStatisticsMixin(AnalysisTool):
+    parameterizedBand: bool = True
+
+    def setDefaults(self):
+        super().setDefaults()
+        self.prep.selectors.flagSelector = CoaddPlotFlagSelector()
+        self.prep.selectors.snSelector = SnSelector(fluxType="{band}_psfFlux", threshold=100)
+
+        self.process.buildActions.patchWhole = LoadVector()
+        self.process.buildActions.patchWhole.vectorKey = "patch"
+
+        self.process.buildActions.mags = MagColumnNanoJansky(vectorKey="{band}_psfFlux")
+        # pre-compute a stellar selector mask so it can be used in the filter
+        # actions while only being computed once, alternatively the stellar
+        # selector could be calculated and applied twice in the filter stage
+        self.process.buildActions.starSelector = PsfStarSelector()
+
+        self.process.filterActions.xStars = DownselectVector(
+            vectorKey="mags", selector=VectorSelector(vectorKey="starSelector")
+        )
+        # downselect the psfFlux as well
+        self.process.filterActions.psfFlux = DownselectVector(
+            vectorKey="{band}_psfFlux", selector=VectorSelector(vectorKey="starSelector")
+        )
+        self.process.filterActions.psfFluxErr = DownselectVector(
+            vectorKey="{band}_psfFluxErr", selector=VectorSelector(vectorKey="starSelector")
+        )
+
+        self.process.filterActions.patch = DownselectVector(
+            vectorKey="patchWhole", selector=VectorSelector(vectorKey="starSelector")
+        )
+
+        self.process.calculateActions.stars = CalcRhoStatistics()
+
+        self.process.calculateActions.stars.treecorr.nbins = 10
+        self.process.calculateActions.stars.treecorr.min_sep = 0.1
+        self.process.calculateActions.stars.treecorr.max_sep = 100.0
+        self.process.calculateActions.stars.treecorr.sep_units = "arcmin"
+        self.process.calculateActions.stars.treecorr.metric = "Arc"
```

Comment:
Don't use Arc.  It's just slower and doesn't add anything useful compared to the default Euclidean.